### ISSUE TITLE
Date created on media files set to date of export instead of original date created

### ISSUE THREAD

**Issue by pimduin** [KnugiHK/WhatsApp-Chat-Exporter#65]
In the resulting export the "date created" on each file is set to the date the exporter is run.

In case of the media files, is it possible for these files to retain their original "date created" dates after an export?

**Comment by KnugiHK** [comment]
Hi. Are you working on Android or iOS?

It is possible to change the modify and access date with `os.utime` easily. However, the creation date seems to requires some extra efforts or libraries. I will look into it.

**Comment by pimduin** [comment]
I'm running the tool on an iOS backup. 
Thank you for looking into it.

**Comment by KnugiHK** [comment]
You can now try the `--preserve-timestamp` flag, which will preserve the modification time of the extracted media files.

As for creation time, it is very platform-dependent and may require other dependencies. Therefore, it may or may not be implemented, given that the feature is not the main focus of the exporter.

**Comment by pimduin** [comment]
I just tested it out, it did not work at first on an encrypted iOS backup. When using an unencrypted backup the new command line option seems to work! Thanks a lot! 

I'm in the process of archiving parts of the whatsapp media from the backup to my photo collection software. Having a proper creation date is a huge help in this case. 

I will be going though all media eventually, if I happen to notice something off I will report it.


By the way, this is the error I get when using an encrypted iOS backup:

```
* Command:
wtsexporter -i --preserve-timestamp -b ./backup_folder

* Output:
Encryption detected on the backup!
Enter the password for the backup:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/bin/wtsexporter", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/Whatsapp_Chat_Exporter/__main__.py", line 292, in main
    extract_iphone_media.extract_media(args.backup, identifiers, args.preserve_timestamp)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/Whatsapp_Chat_Exporter/extract_iphone_media.py", line 84, in extract_media
    extract_encrypted(base_dir, password, identifiers)
TypeError: extract_encrypted() missing 1 required positional argument: 'preserve_timestamp'

```

**Comment by KnugiHK** [comment]
> I just tested it out, it did not work at first on an encrypted iOS backup. When using an unencrypted backup the new command line option seems to work! Thanks a lot!

Yes. It has yet to be implemented on encrypted backup, as it involves another [module](https://github.com/KnugiHK/iphone_backup_decrypt).

> By the way, this is the error I get when using an encrypted iOS backup:

However, regarding the error you encountered, it should fixed with 380289d.

**Comment by KnugiHK** [comment]
As per 8eb0542, the timestamp should also be preserved when working with encrypted backup.

**Comment by Box333** [comment]
Hi KnugiHK,

thank you for you for adding my request.

i am trying to extract the CameraRoll Domain with iphone_backup_dectrypt only (not using whatsapp_extract).

Is it possible to add the --preserve-timestamp switch also to iphone_backup_dectrypt script itself, so that we can use that switch directly ?

Thanks a lot

**Comment by KnugiHK** [comment]
Released in [0.9.7](https://github.com/KnugiHK/WhatsApp-Chat-Exporter/releases/tag/0.9.7).

**Comment by KnugiHK** [comment]
@filipcoja You requested feature is merged! See #194, which will be released in the next version.
